pipeline/data/download-mono.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+"""
+Downloads a monolingual dataset, shuffles it, and truncates it to a maximum amount of sentences.
+
+Kinds:
+   taskcluster/kinds/dataset/kind.yml
+
+Example usage:
+
+    pipeline/data/download-mono.py                  \\
+        --dataset news-crawl_news.2021              \\
+        --language en                               \\
+        --max_sentences 100000000                   \\
+        --artifacts $TASK_WORKDIR/artifacts
+
+Artifacts:
+
+    artifacts
+    └── news.2021.en.zst
+"""
+
+import argparse
+import gzip
+import io
+import os
+import re
+from random import Random
+from typing import Iterator, Optional
+
+import requests
+import zstandard
+from google.cloud.storage.fileio import BlobReader
+
+from pipeline.utils import Dataset, attempt_mocked_request
+from pipeline.utils.downloads import google_cloud_storage
+from pipeline.utils.logging import get_logger
+
+# TODO(CJK) - Issue #424
+MAX_WORDS_IN_SENTENCE = 100
+
+CURRENT_FOLDER = os.path.dirname(os.path.abspath(__file__))
+IMPORTERS_PATH = os.path.abspath(os.path.join(CURRENT_FOLDER, "mono"))
+
+logger = get_logger("download_mono")
+
+
+def shuffle_and_truncate(
+    line_stream: Iterator[str], dataset: Dataset, file_destination: str, max_sentences: int
+):
+    lines: list[str] = []
+
+    logger.info("Load the stream into memory, discarding sentences that are too long.")
+    for line in line_stream:
+        if len(line.split()) < MAX_WORDS_IN_SENTENCE:
+            lines.append(line)
+
+    logger.info("Perform an in-memory shuffle of the dataset.")
+    random = Random(dataset.key)  # Make this deterministic based on dataset key.
+    random.shuffle(lines)
```

Comment:
this can go OOM on huge datasets

Reply:
merge-mono performs a shuffle, so I could just delete this code.

Reply:
Ok, I looked into this a bit.
Using an 11G file, `shuf` definitely loads it all into memory by default.
<img width="291" alt="image" src="https://github.com/mozilla/firefox-translations-training/assets/1588648/33315b55-a3fa-41da-83d3-34619863a2cb">

But specifying -n limits how much memory is used in total. The documentation stats: `  -n, --head-count=COUNT    output at most COUNT lines`

The memory usage was ~400mb for 1,000,000 limit.

<img width="1042" alt="image" src="https://github.com/mozilla/firefox-translations-training/assets/1588648/9c0d469d-5595-4c25-b437-5c82257ed5b1">


<img width="125" alt="image" src="https://github.com/mozilla/firefox-translations-training/assets/1588648/414b3230-f8fb-4879-8096-566eb4e5266e">

Reply:
It's fine for now while we're working with news-crawl. Let' get back to this when it goes OOM (likely when we add support of common-crawl). Even then we might be able to fix it with just bumping RAM  on a vm.

https://github.com/mozilla/firefox-translations-training/blob/6b55a4b93c2a96040b945e20722e5f81a1f6426e/pipeline/data/download-mono.sh#L37

I believe the idea of this is to take a random sample of the dataset with the specified size. I would preserve this logic but again it's needed for larger mono datasets.

Reply:
This will be resolved via #467